Create useRefs hook (#6)
mediamonks/react-hooks: src/useRefs/useRefs.stories.mdx

> import { Meta } from '@storybook/addon-docs';

Reviewer: We should also include documentation for `arrayRef`

Author: Added docs for `arrayRef`

<Meta title="useRefs/docs" />

# useRefs

> Utility to create an object that automatically create refs. One ref object to rule them all, you

Reviewer: ```suggestion
Utility to create an object that automatically creates refs. One ref object to rule them all, you
```

don't have to use `useRef` for every element that you want to create a reference for.

## Reference

```ts
export function useRefs<T extends Record<string | symbol, RefObject<unknown>>>(
  initialTarget?: Partial<T>,
): T;
```

### Returns

> - `refs` An object that automatically creates refs when accessing it's fields, return type matches

Reviewer: ```suggestion
- `refs` An object that automatically creates refs when accessing its fields. The return type matches
```

  the structure and type of the generics argument.

## Usage

```tsx

> type DemoComponentRefs = Refs<{
>   heading: HTMLHeadingElement;
>   button: Array<HTMLButtonElement | null>;
> }>;
> 
> function DemoComponent() {
>   const refs = useRef<DemoComponentRefs>();
> 
>   console.log(refs.element.current); // HTMLHeadingElement | null
>   console.log(refs.elements.current); // Array<HTMLButtonElement | null>
> 
>   return (
>     <div ref={refs.element}>
>       {items.map((item, index) => (
>         <span key={item} ref={arrayRef(refs.elements, index)}>
>           {item}
>         </span>
>       ))}
>     </div>
>   );
> }

Reviewer: ```suggestion
type DemoComponentRefs = Refs<{
  heading: HTMLHeadingElement;
  buttons: Array<HTMLButtonElement | null>;
}>;

function DemoComponent() {
  const refs = useRef<DemoComponentRefs>();

  console.log(refs.heading.current); // HTMLHeadingElement | null
  console.log(refs.buttons.current); // Array<HTMLButtonElement | null>

  return (
    <div ref={refs.heading}>
      {items.map((item, index) => (
        <span key={item} ref={arrayRef(refs.buttons, index)}>
          {item}
        </span>
      ))}
    </div>
  );
}
```
